core/hyphenator-liang.lua

```diff
@@ -113,23 +113,19 @@ local hyphenateNode = function (node)
   if #segments > 1 then
     local hyphen = SILE.shaper:createNnodes(SILE.settings:get("font.hyphenchar"), node.options)
     local newnodes = {}
-    local leadingApostrophe, nextLeadingApostrophe
     for j, segment in ipairs(segments) do
+      local leadingApostrophe = nil
```

Comment:
```suggestion
      local leadingApostrophe
```

The purpose here is scope not assignment, I think this style says that more clearly.

Reply:
Agreed!